compaction/compaction_manager.cc

```diff
@@ -1183,7 +1183,7 @@ class compaction_manager::validate_sstables_compaction_task : public compaction_
                     sst->get_sstable_level(),
                     sstables::compaction_descriptor::default_max_sstable_bytes,
                     sst->run_identifier(),
-                    sstables::compaction_type_options::make_scrub(sstables::compaction_type_options::scrub::mode::validate));
+                    sstables::compaction_type_options::make_scrub(sstables::compaction_type_options::scrub::mode::validate, _compacting_table->logalloc_tracker()));
```

Comment:
Ho hum. Instead of passing logalloc::tracker and make everyone scratch their head, how about passing a function that makes memtables? That will push the odd dependency to the caller.

We should have hidden the memtable completely in some way, but that's harder and this is really an ad-hoc use case.